src/routes/main.js

```diff
@@ -47,6 +47,9 @@ export default [
         { translation: 'JOINGROUP.ALL_GROUPS', route: { name: 'groupsGallery' } },
         { type: 'activeGroupInfo' },
       ],
+      beforeEnter: 'groups/selectGroupInfo',
+      onError: 'group id not found',
+      onLeave: 'groups/clearGroupInfo',
```

Comment:
`beforeLeave` (or maybe it has to be `afterLeave`) would make it:
a) consistent with vue router lifecycle names
b) clearer what it does * (it can never do it exactly ON route, only just before, or just after)

\* if combined with my suggestion below :)

Reply:
`beforeLeave` seems good, the question is if the action is allowed to abort the transition? I mostly intended it to be a clear-up task, so `afterLeave` should be fine as well.